engine/runtime/src/main/java/org/enso/interpreter/EnsoLanguage.java

```diff
@@ -370,4 +393,12 @@ protected Object getLanguageView(EnsoContext context, Object value) {
     }
     return null;
   }
+
+  public ExecutionEnvironment getThreadExecutionEnvironment() {
+    return threadExecutionEnvironment.get().getExecutionEnvironment();
```

Comment:
The method should be renamed to `getExecutionEnvironment`. In my opinion it should:
```java
var env = threadExecutionEnvironment.get().getExecutionEnvironment();
return env != null ? env : getGlobalExecutionEnvironment();
```